python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -193,6 +195,26 @@ def get_empty_subset(self, *, key: T_EntityKey) -> EntitySubset[T_EntityKey]:
         value = partitions_def.empty_subset() if partitions_def else False
         return EntitySubset(self, key=key, value=_ValidatedEntitySubsetValue(value))
 
+    def get_asset_subset_from_asset_graph_subset(
+        self, asset_graph_subset: AssetGraphSubset, asset_key: AssetKey
+    ) -> Optional[EntitySubset[AssetKey]]:
+        return self.get_subset_from_serializable_subset(
+            asset_graph_subset.get_asset_subset(asset_key, self.asset_graph)
+        )
+
+    def iterate_asset_subsets(
+        self, asset_graph_subset: AssetGraphSubset
+    ) -> Iterable[EntitySubset[AssetKey]]:
+        """Returns an Iterable of EntitySubsets representing the subset of each asset that this
+        AssetGraphSubset contains.
+        """
+        for serializable_entity_subset in asset_graph_subset.iterate_asset_subsets(
+            self.asset_graph
+        ):
+            yield check.not_none(
```

Comment:
I wonder if we need some kind of object that represents a validated AssetGraphSubset in the same way that EntitySubset is a validated EntitySubsetValue? Then we could do the validation before anything comes into this function instead of potentially in the middle of the iteration

Reply:
for now I took your good advice to make get_entity_subset_from_asset_graph_subset (formerly get_asset_subset_from_asset_graph_subset) never return None, added more clear validation errors there, and delegated to that method here